test/test-app/cfg.test.lua

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env tarantool
+
+-- Test is an example of TAP test
+
+local tap = require('tap')
+local test = tap.test('cfg')
+test:plan(4)
+
+box.cfg{listen = box.NULL}
+test:is(nil, box.info.listen, 'no cfg.listen - no info.listen')
+
+box.cfg{listen = '127.0.0.1:0'}
+test:ok(box.info.listen:match('127.0.0.1'), 'real IP in info.listen')
+test:ok(not box.info.listen:match(':0'), 'real port in info.listen')
+
+box.cfg{listen = box.NULL}
+test:is(nil, box.info.listen, 'cfg.listen reset drops info.listen')
+
+test:check()
+os.exit(0)
```

Comment:
The [rule of thumb](https://github.com/tarantool/doc/issues/1004) is to set a non-zero exit code, when the check for the plan fails.

Reply:
Updated.

Reply:
`os.exit(<boolean>)` always gives zero exit code. I suggest to use `os.exit(test:check() and 0 or 1)`.

Reply:
```
--- a/test/test-app/cfg.test.lua
+++ b/test/test-app/cfg.test.lua
@@ -16,4 +16,4 @@ test:ok(not box.info.listen:match(':0'), 'real port in info.listen')
 box.cfg{listen = box.NULL}
 test:is(nil, box.info.listen, 'cfg.listen reset drops info.listen')
 
-os.exit(test:check())
+os.exit(test:check() and 0 or 1)
```